addons/account/models/account_move.py

```diff
@@ -500,6 +507,10 @@ def _onchange_partner_id(self):
         bank_ids = self.bank_partner_id.bank_ids.filtered(lambda bank: bank.company_id is False or bank.company_id == self.company_id)
         self.partner_bank_id = bank_ids and bank_ids[0]
 
+        # Recompute 'partner_shipping_id' based on 'partner_id'.
+        addr = self.partner_id.address_get(['delivery'])
+        self.partner_shipping_id = addr and addr.get('delivery')
+
         # Find the new fiscal position.
         delivery_partner_id = self._get_invoice_delivery_partner_id()
```

Comment:
@casp-odoo We might not need the method _get_invoice_delivery_partner_id anymore or at least it should take into account the delivery address in account already (this method was originally done to be inherited in sale because the delivery address was added over there).  It is used in a lot of places however.

Reply:
Yes, this should indeed be replaced by `partner_shipping_id` everywhere it is used